### ISSUE TITLE
Stabilization fit ignored on options change

### ISSUE THREAD

**Issue by rjbriody** [almende/vis#1708]
Hello,

In the following example you will notice that stabilization fit happens on the initial draw, but when the options are changed the layout/physics are "all springy and stuff".

https://jsfiddle.net/h9fv04yb/

I was expecting the graph to draw without physics animation when double-clicking the empty canvas. Interestingly, if I call ```network.setData(data)``` after changing the options, then it works as expected, i.e. it draws without physics animation.

**Comment by paschsaar** [comment]
I face the same problem. With the old version of vis.js I was able to update the Dataset like to hide some nodes and edges without triggering of redraw or triggering stabilisation. I just blended the hidden elements out. With the latest version I am not able to do so. With each update of dataset is is triggering the stabilisation, even with deactivated stabilisation. 
I just face this problem with hierarchical layout sortmethod directed.

@rjbriody Your hack with network.setData(data) does not work for me.

**Comment by AlexDM0** [comment]
Hi,

The "all springy and stuff" is not an animation. This is the physics stabilizing. If you want to hide this from the user, call network.stabilize after you disable the hierarchical view. Read up on what exactly is happening in the documentation. 

@paschsaar the network has always redrawn and restabilized on a change in any layout. If you can reproduce it in a jsbin so I can see what you're doing m, that would be great. It could be that very old versions always trigger the hidden stabilization. 

Regards

**Comment by paschsaar** [comment]
Hello AlexDM0,

this is a part of my code: 
[(https://jsfiddle.net/w1bctxuL/6/)]

With the old version i was able to click on one node and the node stayed at the same place. No physics and redraw. The same network but with some hidden nodes and edges

**Comment by AlexDM0** [comment]
Hi,

Which old version are you referring to?

Regards

**Comment by paschsaar** [comment]
This is the example with the old version. 
https://jsfiddle.net/5t2xmjvs/
Old = 4.8.0

**Comment by AlexDM0** [comment]
Right, my bad. With hiding, I thought you meant removing them from the dataset. I'll look into this.

Regards

**Comment by AlexDM0** [comment]
I see the cause is here:
https://github.com/almende/vis/blob/master/lib/network/modules/EdgesHandler.js#L189

I'm not 100% sure why this was added but I believe it fixed a bug related to clustering... I'm not sure when I'll get around to fixing this though as it would be a lot of testing..

**Comment by paschsaar** [comment]
Thank you for investigating into it. 
Are you going to trigger me if there is a solution?

**Comment by AlexDM0** [comment]
If this is fixed this issue will be updated and closed when we release the fix so you'll be notified by git :)

**Comment by paschsaar** [comment]
Can you give me a timeframe? I would also use a nonreleased version.

**Comment by AlexDM0** [comment]
Sorry, I cant promise anything at the moment

**Comment by paschsaar** [comment]
Hi Alex,

are there some updates for me?

**Comment by AlexDM0** [comment]
Yes, still busy and I haven't gotten around to it. Feel free to implement it yourself though ;)

**Comment by paschsaar** [comment]
Hello Alex,

I took a look into the code but I do not found the problem. This project is that huge, it would need to much time for me to investigate further. 
Maybe one hint from my side: This wrong behaviour appears in the version  4.11.0. Until 4.10.0 it works fine. 
I am not a private developer and I do not have to much time for this task.

Thank you for your great work! It helps us a loot in out daily tasks and the whole department likes to see it work again.

**Comment by paschsaar** [comment]
I will go as a temporary solution for the version 4.10.0
Anyhow I like the new features for the physic and I am waiting for a new release with a bugfix

**Comment by AlexDM0** [comment]
Seems like a good plan. I can't give you an eta on the fix but you can try this: change the colors to rgba(0,0,0,0) instead of using hidden true. 

It's a little ugly but it should do it well enough. 

Cheers

**Comment by paschsaar** [comment]
Hello Alex,

I just saw that you released a new version since our last conversation. I was not able to find out if the finding above is part of it?

**Comment by AlexDM0** [comment]
Hi,

It's not. Development on vis is very slow now since both Jos and I have new jobs. If you'd like to look into this issue and provide a pull req that would be great.

Regards
